### DISCUSSION TITLE
Is it possible to trade off range with precision?

### DISCUSSION THREAD

**Discussion by pruj-speed** [CadQuery/cadquery#1702] [Q&A] [answered]
Hi Folks,

I work in semiconductors where layer thickness are sometimes of the order of a few nanometers. I notice that when I extrude a layer that is only 0.1µm thick then I run into an error. My assemblies will at most be a few centimeters.

x,y,z = (10E-6,10E-6,1E-7)
substrate = cq.Workplane("XY").box(x, y, z, centered=True)
show(substrate)

Error: Standard Domain Error
```python
class BRepPrimAPI_MakeBox(OCP.BRepBuilderAPI.BRepBuilderAPI_MakeShape, OCP.BRepBuilderAPI.BRepBuilderAPI_Command):
    """
    Describes functions to build parallelepiped boxes. A MakeBox object provides a framework for: - defining the construction of a box, - implementing the construction algorithm, and - consulting the result. Constructs a box such that its sides are parallel to the axes of - the global coordinate system, or - the local coordinate system Axis. and - with a corner at (0, 0, 0) and of size (dx, dy, dz), or - with a corner at point P and of size (dx, dy, dz), or - with corners at points P1 and P2. Exceptions Standard_DomainError if: dx, dy, dz are less than or equal to Precision::Confusion(), or - the vector joining the points P1 and P2 has a component projected onto the global coordinate system less than or equal to Precision::Confusion(). In these cases, the box would be flat.
```

Is there a way to simply trade off the range with precision in cadquery? Kind of like what is being discussed here https://gdspy.readthedocs.io/en/stable/gettingstarted.html#units-in-gdsii

**Comment by adam-urbanczyk** [comment, marked as answer]
CQ is unitless, so just work in in natural units for your scale.

**Reply by pruj-speed** [reply]
thank you. I will do that
